pkg/controlplane/apiserver/options/options.go

```diff
@@ -229,7 +229,9 @@ func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (Comple
 	}
 
 	// put authorization options in final state
-	completed.Authorization.Complete()
+	if completed.Authorization != nil {
+		completed.Authorization.Complete()
```

Comment:
In other, similar cases, we check for nil in the method.

Reply:
TBH not sure what you mean. the only thing above can be setting strict struct empty as its nil'able?

Reply:
Internet quick search says - nobody is using it :D

Reply:
see line 196
```
func (o *Options) Complete(alternateDNS []string, alternateIPs []net.IP) (CompletedOptions, error) {
	if o == nil {
		return CompletedOptions{completedOptions: &completedOptions{}}, nil
	}
```

you should add the logic here https://github.com/kubernetes/kubernetes/blob/1f40e0916ec41848265b510d15a14be308f9dc60/pkg/kubeapiserver/options/authorization.go#L91-L97